src/auth.rs

```diff
@@ -87,3 +90,166 @@ impl AppAuth {
         create_jwt(self.app_id, &self.key).context(crate::error::JWTSnafu)
     }
 }
+
+/// The data necessary to authenticate as a github OAtuh app.
+#[derive(Clone, Deserialize)]
+#[serde(from = "OAuthWire")]
+pub struct OAuth {
+    pub access_token: SecretString,
+    pub token_type: String,
+    pub scope: Vec<String>,
+}
+
+/// The wire format of the OAuth struct.
+#[derive(Deserialize)]
+struct OAuthWire {
+    access_token: String,
+    token_type: String,
+    scope: String,
+}
+
+impl From<OAuthWire> for OAuth {
+    fn from(value: OAuthWire) -> Self {
+        OAuth {
+            access_token: SecretString::from(value.access_token),
+            token_type: value.token_type,
+            scope: value.scope.split(",").map(ToString::to_string).collect(),
+        }
+    }
+}
+
+/// Authenticate with Github's device flow. This starts the process to obtain a new `OAuth`.
+///
+/// See https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps#device-flow for details.
+pub async fn authenticate_as_device<I, S>(client_id: SecretString, scope: I) -> Result<DeviceCodes>
```

Comment:
This should not be a free function, it should be an inherent method on `Octocrab`, otherwise this code won’t work with enterprise installations which use different base URLs.

Reply:
I wanted a way to do this without requiring the user to reconfigure a second builder. I now see [Octocrab::builder](https://docs.rs/octocrab/latest/octocrab/struct.Octocrab.html#implementations). I'll make this change.